cache/httpproxy/caching_http_proxy.go

```diff
@@ -37,15 +37,16 @@ type CacheableResponse struct {
 
 func (p cachingHttpProxy) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	// Rewrite the request URL to new host
-	targetURL, _ := url.JoinPath(p.targetHost, req.URL.RequestURI())
```

Comment:
`req.URL.RequestURI()` returns the escaped url which means the query is displayed as `/files/contracts/any/42262%3Fpage=0`, which when used as input to `url.JoinPath` causes the query param to be melded into the chain and then sourcify complains that there's no chain called `42262?page=0`

Reply:
weird, https://pkg.go.dev/net/url#URL.RequestURI that's not documented to be escaped. was it actually %3F in the original URL?

Reply:
Huh that's strange. I think the issue is elsewhere though since the proxy server constructs a new `url` object, perhaps the query params aren't being properly parsed there somehow.

> was it actually %3F in the original URL?
In the sourcify client it's encoded properly as `&` and the params themselves are added via `url.RawQuery` but when it gets to the proxy server the `&` was escaped with `%3F`.

Reply:
`%3F` is `?`, `%26` is `&`